dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/metadata/inputRenderer/suggestion/GrantSuggestionHandler.java

```diff
@@ -0,0 +1,47 @@
+package edu.harvard.iq.dataverse.dataset.metadata.inputRenderer.suggestion;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+
+@Stateless
+public class GrantSuggestionHandler implements SuggestionHandler {
+
+    private GrantSuggestionDao grantSuggestionDao;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    @Deprecated
+    public GrantSuggestionHandler() {
+    }
+
+    @Inject
+    public GrantSuggestionHandler(GrantSuggestionDao grantSuggestionDao) {
+        this.grantSuggestionDao = grantSuggestionDao;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation always returns class name.
+     */
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation generates suggestions with filters if there is something in map, or without filters if map is empty.
+     */
+    @Override
+    public List<String> generateSuggestions(Map<String, String> filters, String suggestionSourceFieldName, String suggestionSourceFieldValue) {
+
+        return filters.isEmpty() ? grantSuggestionDao.fetchSuggestions(suggestionSourceFieldName, suggestionSourceFieldValue, 10) :
+                grantSuggestionDao.fetchSuggestions(filters, suggestionSourceFieldName, suggestionSourceFieldValue, 10);
```

Comment:
Yes but then the DAO method would had to be bigger, I wanted to avoid any logic handling in DAO since it is meant to just query from db. Also if you wanted to use method (in the future somewhere) without filters you would had to pass empty map which is kinda counter intuitive. It is more logical IMO to have a method without filter parameter if you don't have any filters.

Reply:
You can just add `if` logic and inside call method without filters if filters are empty. And since we are not using it elsewhere for now, make it private so in case we need that in the future we can just make it public if needed.

Not being able to pass empty filters list is error prone.

Reply:
I think you are tackling 2 problems here. 1. Empty Map - I think I should just add check for empty map and return empty list. 2. Method modification - that would violate 'Open–closed principle' if we can design something in the way that we don't have to modify it in the future, then we won't break the principle and other programmers will have less work to do.

Reply:
When passed map is empty OR resolves to empty filters I expect method to perform normal, unfiltered fetching.